src/libsystemd-network/dhcp-option.c

```diff
@@ -93,32 +112,74 @@ static int option_append(uint8_t options[], size_t size, size_t *offset,
 
                 options[*offset] = code;
                 options[*offset + 1] = l;
-
                 *offset += 2;
 
                 ORDERED_SET_FOREACH(p, s) {
-                        options[*offset] = p->option;
-                        options[*offset + 1] = p->length;
-                        memcpy(&options[*offset + 2], p->data, p->length);
-                        *offset += 2 + p->length;
+                        assert_se(dhcp_option_append_tlv(options, size, offset, p->option, p->length, p->data) >= 0);
```

Comment:
Hmm, of course this should not fail, but `dhcp_option_append_tlv()` may be updated in the future. So, I think it is better to use the usual error handling way, that is,
```
r = func(...);
if (r < 0)
        return r;
```